### PR TITLE
Emc workflow

### PR DESCRIPTION
**Pull request by svandenb-dev**

### CODE REVIEW

File: pyaedt/edb_core/EDB_Data.py [hunk 1]
```diff
@@ -5930,3 +5931,435 @@ def num_resistors(self):
...
+    @property
+    def conductivity(self):
+        return self._conductivity
+
+    @conductivity.setter
+    def conductivity(self, value):
+        self._conductivity = value
+
+    def _json_format(self):
+        """Export Json file from SimulationConfiguration object.
```
Comment (PipKat) [inline pyaedt/edb_core/EDB_Data.py:6079]:
```suggestion
        """Export JSON file from ``SimulationConfiguration`` object.
```

File: pyaedt/edb_core/EDB_Data.py [hunk 2]
```diff
@@ -5930,3 +5931,435 @@ def num_resistors(self):
...
+
+        Parameters
+        ----------
+        output_file : str
+            Json file name.
```
Comment (PipKat) [inline pyaedt/edb_core/EDB_Data.py:6084]:
```suggestion
            Name of JSON file.
```

File: pyaedt/edb_core/EDB_Data.py [hunk 3]
```diff
@@ -5930,3 +5931,435 @@ def num_resistors(self):
...
+
+        Returns
+        -------
+        bool
+            True when succeeded False when file name not provided.
```
Comment (PipKat) [inline pyaedt/edb_core/EDB_Data.py:6089]:
```suggestion
            ``True`` when successful, ``False`` when file name is not provided.
```

File: pyaedt/edb_core/stackup.py [hunk 4]
```diff
@@ -133,6 +134,22 @@ def material(self, name):
         layer_clone.SetMaterial(name)
         self._pclass._set_layout_stackup(layer_clone, "change_attribute")
 
+    @property
+    def filling_material(self):
+        """Get/Set  the filling material.
```
Comment (PipKat) [inline pyaedt/edb_core/stackup.py:139]:
```suggestion
        """Filling material.
```

File: pyaedt/edb_core/stackup.py [hunk 5]
```diff
@@ -133,6 +134,22 @@ def material(self, name):
...
+
+        Returns
+        -------
+        Edb material.
```
Comment (PipKat) [inline pyaedt/edb_core/stackup.py:143]:
```suggestion
        EDB material.
```

File: pyaedt/edb_core/stackup.py [hunk 6]
```diff
@@ -754,6 +771,24 @@ def export_stackup(self, fpath, file_format="csv"):
             df.to_excel(fpath)
         return True
 
+    def check_and_fix_stackup(self, stackup_info=None):
+        if not isinstance(stackup_info, StackupInfo):
+            logger.error("Argument must be an instance of StackInfo")
```
Comment (PipKat) [inline pyaedt/edb_core/stackup.py:776]:
```suggestion
            logger.error("Argument must be an instance of StackInfo.")
```

File: pyaedt/edb_core/stackup.py [hunk 7]
```diff
@@ -754,6 +771,24 @@ def export_stackup(self, fpath, file_format="csv"):
...
+        signal_layers = [layer for layer in stackup_info.layers["primary"] if layer.type == "conductor"]
+        dielectric_layers = [layer for layer in stackup_info.layers["primary"] if layer.type == "dielectric"]
+        for signal_layer in signal_layers:
+            layer_to_check = self.layers[signal_layer.name]
+            if layer_to_check:
+                layer_to_check.filling_material = dielectric_layers[1].material_name
+
 
 class EdbStackup(object):
     """Manages EDB methods for stackup and material management accessible from `Edb.core_stackup` property.
```
Comment (PipKat) [inline pyaedt/edb_core/stackup.py:794]:
```suggestion
    """Manages EDB methods for stackup and material management accessible from ``Edb.core_stackup`` property.
```

File: pyaedt/edb_core/siwave.py [hunk 8]
```diff
@@ -1435,6 +1435,75 @@ def _setup_decade_count_sweep(self, sweep, start_freq, stop_freq, decade_count):
...
+        component_name:
+            str, component name.
+        pins_name:
+            str, pin name or list of pin name.
+        layer_name:
+            str, layer name
+        reference_net:
+            str, reference net name.
+        impedance:
+            float, port impedance.
```
[the remark this answers is not in the document]
Comment (MaxJPRey) [reply pyaedt/edb_core/siwave.py:1454]: Thanks @PipKat for improving my suggestion 👍 .

[the remark this answers is not in the document]
Comment (MaxJPRey) [reply pyaedt/edb_core/siwave.py:1454]: @svandenb-dev This should fix most of the issue regarding the failing "Documentation" action.

File: pyaedt/edb_core/siwave.py [hunk 9]
```diff
@@ -1435,6 +1435,75 @@ def _setup_decade_count_sweep(self, sweep, start_freq, stop_freq, decade_count):
             freq = freq * math.pow(10, 1.0 / decade_cnt)
             sweep.Frequencies.Add(str(freq))
 
+    def create_port_between_pin_and_layer(
+        self, component_name=None, pins_name=[], layer_name=None, reference_net=None, impedance=50.0
+    ):
+        """
```
Comment (PipKat) [inline pyaedt/edb_core/siwave.py:1441]:
```suggestion
        """Create a port between a pin and a layer.
```

**Pull request closed without merging**